logging/e2e/cloudrun.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash -e
+
+# Copyright 2020 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# GCP permissions needed:
+# APIs
+# - cloudbuild.googleapis.com
+# - run.googleapis.com
+# ROLES
+# - enable pub/sub to create authentication tokens   
+# - storage.buckets.create for CloudBuild service account
+
+set -eo pipefail
+
+GCR_REGION=us-west1
+PROJECT_ID=$(gcloud config get-value project 2> /dev/null)
+TOPIC_ID=$2
+
+scaffold() {
+    # Gets folder containing this running script
+    SCRIPT_DIR=$(realpath $(dirname "$0"))  # e2e/.
+    ROOT_DIR=$SCRIPT_DIR/../..              # google-cloud-go/.
+
+    # Handle if e2e_test didn't create topic
+    missing=$(gcloud pubsub topics describe $TOPIC_ID 2>&1) || true
+    if [[ "$missing" == *"NOT_FOUND"* ]]; then
+        gcloud pubsub topics create $TOPIC_ID
+    else
+        echo TOPIC FOUND
+    fi
+
+    # Temporarily move Dockerfile into the local library build context
+    cd $ROOT_DIR
+    cp ./logging/e2e/Dockerfile .
+
+    # Build latest test app image
+    gcloud builds submit --tag gcr.io/${GCLOUD_TESTS_GOLANG_PROJECT_ID}/$TOPIC_ID
+    rm Dockerfile
+
+    # Build test image to Cloud Run
+    gcloud run deploy --image gcr.io/${GCLOUD_TESTS_GOLANG_PROJECT_ID}/$TOPIC_ID \
+        --platform managed \
+        --region $GCR_REGION \
+        --allow-unauthenticated \
+        --set-env-vars=TOPIC_ID=$TOPIC_ID,ENVIRONMENT="cloudrun" \
+        $TOPIC_ID
+
+    # Allow Pub/Sub to create authentication tokens in your project:
+    PROJECT_NUMBER=$(gcloud projects describe log-bench --format 'value(projectNumber)')
```

Comment:
did you mean PROJECT_ID instead of log-bench?

Reply:
ah good catch